### DISCUSSION TITLE
能否动态创建类并继承泛型父类并编译

### DISCUSSION THREAD

**Discussion by huangyong2016** [dotnetcore/Natasha#229] [Q&A]
public class FrontendApiService : DataService<IFrontendApiBLL, Dictionary<string, object>>{}

**Comment by huangyong2016** [comment]
DataService是继承了ControllerBase，编译会报异常：NatashaException: 类型“ControllerBase”在未引用的程序集中定义。必须添加对程序集“Microsoft.AspNetCore.Mvc.Core, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60”的引用。

**Reply by NMSAzulX** [reply]
@huangyong2016 是引用问题，参照文档覆盖引用。https://natasha.dotnetcore.xyz/zh-Hans/docs/api/Preheating

**Comment by huangyong2016** [comment]
你所发的邮件已收到，查阅后再回复！

**Comment by huangyong2016** [comment]
我想实现动态编译webapi接口程序集，builder.Add(DefaultUsing.UsingScript + code);的时候会异常提示：

NatashaException: 应输入标识符
AssemblyCSharpBuilder.Add(string script)


已引入包：
DotNetCore.Natasha.CSharp
DotNetCore.Compile.Environment

代码如下：

               NatashaInitializer.Preheating((asmName, name) => {
                    if (name != null)
                    {
                        return false;
                    }
                    return true;
                });

                AssemblyCSharpBuilder builder = new();
                builder.ConfigCompilerOption(item => item.AddSupperess("CS0108"));
 
                builder.Domain = NatashaManagement.CreateRandomDomain();

              
                builder.Add(DefaultUsing.UsingScript + @"public class A{}");
                //获取编译后的动态程序集
                var assembly = builder.GetAssembly();

**Reply by NMSAzulX** [reply]
这个提示一般是缺少标点符号，你可以把脚本放到.cs文件里看一下具体错误。

**Reply by NMSAzulX** [reply]
@huangyong2016 可以加 V 进群，NMSAzulX
